Labeling service instance response

andreaAlkalay opened the pull request.

royclarkson commented:
@andreaAlkalay thanks for the PR! We prefer to squash commits in PRs into a single commit so it's easier to track changes for a specific issue. Are you familiar with interactive rebasing in git? If you are, can you squash these commits and then force push back to your GitHub branch. If not, I can do it before merging. 

If you fix the commits, please edit the commit message to be more specific about the changes. You can use the information in the original issue as a model. Including that info in the commit message means we don't have to refer back to GitHub issues in order to see the details. If you view the git log, you can see other examples. [This page](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html) has good info about how to author a well formatted commit message. 

Lastly, at the end of the commit message, please indicate which issue is being closed. e.g. `closes #311`. Again, this helps tie everything together, and also will automatically close the issue when this PR is merged. Let me know if you need any help.

andreaAlkalay commented:
> @andreaAlkalay thanks for the PR! We prefer to squash commits in PRs into a single commit so it's easier to track changes for a specific issue. Are you familiar with interactive rebasing in git? If you are, can you squash these commits and then force push back to your GitHub branch. If not, I can do it before merging.
> 
> If you fix the commits, please edit the commit message to be more specific about the changes. You can use the information in the original issue as a model. Including that info in the commit message means we don't have to refer back to GitHub issues in order to see the details. If you view the git log, you can see other examples. [This page](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html) has good info about how to author a well formatted commit message.
> 
> Lastly, at the end of the commit message, please indicate which issue is being closed. e.g. `closes #311`. Again, this helps tie everything together, and also will automatically close the issue when this PR is merged. Let me know if you need any help.

I think I'm done, added the issue description as the commit message and squashed all the commits into a single one.

royclarkson commented on spring-cloud-open-service-broker-core/src/main/java/org/springframework/cloud/servicebroker/model/ServiceInstanceMetadata.java:
```diff
@@ -0,0 +1,161 @@
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.servicebroker.model;
```
This should be in the `org.springframework.cloud.servicebroker.model.instance` package instead.

andreaAlkalay replied:
Done. :)

royclarkson replied:
Thanks!

royclarkson reviewed (changes requested):
@andreaAlkalay thanks for the submission! There are a few items that need to be further updated. The `ServiceInstanceMetadata` needs to be moved into the appropriate `instance` package. Could you also add a `ServiceInstanceMetadataTest` with some basic unit tests? Lastly, I see in the [spec](https://github.com/openservicebrokerapi/servicebroker/blob/v2.16/spec.md#body-7) that this metadata can also be returned in a request to update a service instance. So, we'll need a similar change in `UpdateServiceInstanceResponse` and the corresponding test. Thanks again!

royclarkson commented:
@andreaAlkalay I'm happy with the PR as it is. Thanks for working through the requested changes. I am planning to release a version 3.3 that is based on Spring Boot 2.4 before I merge in these changes. I'm considering the SCOSB 3.4 release to update to OSBAPI 2.16, and I'll merge in this PR once the master branch is ready for that work.

andreaAlkalay commented:
> @andreaAlkalay I'm happy with the PR as it is. Thanks for working through the requested changes. I am planning to release a version 3.3 that is based on Spring Boot 2.4 before I merge in these changes. I'm considering the SCOSB 3.4 release to update to OSBAPI 2.16, and I'll merge in this PR once the master branch is ready for that work.

Do you have a rough time estimation when can I expect to have a release including my changes?
Thanks a lot!

royclarkson commented:
I can publish an initial milestone (3.4.0-M1) that has these changes relatively soon, but a final GA release will need to have all the OSBAPI 2.16 updates, so that it is compatible with the full list of changes in the spec release.

andreaAlkalay commented:
> I can publish an initial milestone (3.4.0-M1) that has these changes relatively soon, but a final GA release will need to have all the OSBAPI 2.16 updates, so that it is compatible with the full list of changes in the spec release.

we're really waiting for this change so if you can release a milestone version it would be great!

royclarkson commented:
> we're really waiting for this change so if you can release a milestone version it would be great!

That's great to hear! I appreciate the interest in the project. :)

andreaAlkalay commented:
> > we're really waiting for this change so if you can release a milestone version it would be great!
> 
> That's great to hear! I appreciate the interest in the project. :)
Hi,
Do you have any estimations on the release of 3.4.0-M1, or any other release which will include my PR?
Thanks!

royclarkson commented:
Closed in b4bbd336bdfcefd0a6afd994e6d52693fd9d26ca

royclarkson commented:
@andreaAlkalay this will be available in 3.4.0-SNAPSHOT

royclarkson commented:
@andreaAlkalay this is now released in [3.4.0-M1](https://github.com/spring-cloud/spring-cloud-open-service-broker/releases/tag/v3.4.0-M1)
